### PR TITLE
feat: add add role component

### PR DESCRIPTION
**Pull request by punwai**
## Description
Have added a new add role component. The component is supposed to link to a view roles page, however, this has not yet been created. I will create this in a following PR.

## Related issues and discussion
https://github.com/openMF/web-app/issues/487

## Screenshots, if any

Mp4

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/system/system-routing.module.ts [hunk 1]
```diff
@@ -221,11 +222,21 @@ const routes: Routes = [
       },
       {
           path: 'roles-and-permissions',
-          component: RolesAndPermissionsComponent,
           resolve: {
                 roles: RolesAndPermissionsResolver
           },
```
[the remark this answers is not in the document]
Comment (punwai) [reply src/app/system/system-routing.module.ts]: Have added change

File: src/app/system/roles-and-permissions/add-role/add-role.component.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  styleUrls: ['./add-role.component.scss']
+})
+export class AddRoleComponent implements OnInit {
+
+  /** Role form. */
+  roleForm: FormGroup;
+
+  /**
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService Accounting Service.
```
Comment (abhaychawla) [inline src/app/system/roles-and-permissions/add-role/add-role.component.ts]: Service incorrectly mentioned as accounting here.

File: src/app/system/system-routing.module.ts [hunk 3]
```diff
@@ -221,11 +222,21 @@ const routes: Routes = [
...
-                roles: RolesAndPermissionsResolver
-          },
           data: { title:  extract('Roles and Permissions'), breadcrumb: 'Roles and Permissions' },
+          children: [
+            {
+              path: '',
+              component: RolesAndPermissionsComponent,
+              resolve: {
+                roles: RolesAndPermissionsResolver
+            }
```
Comment (abhaychawla) [inline src/app/system/system-routing.module.ts]: Ensure proper indentation.

Comment (punwai) [reply src/app/system/system-routing.module.ts]: After you commented about the Accounting Service, I went to check all the other components and many of them also has the comment as Accounting Service. I've changed the ones with this to System Service in the new push.

File: src/app/system/roles-and-permissions/add-role/add-role.component.ts (hunk 2 above)
Comment (punwai) [reply src/app/system/roles-and-permissions/add-role/add-role.component.ts]: After you commented about the Accounting Service, I went to check all the other components and many of them also has the comment as Accounting Service. I've changed the ones with this to System Service in the new push.

Comment (abhaychawla) [reply src/app/system/roles-and-permissions/add-role/add-role.component.ts]: Oh, good catch. But don't change them in this PR. This PR should only be for the addition of add role component.

**Pull request merged**